src/lib/message-router.js

```diff
@@ -132,8 +132,10 @@ MessageRouter.prototype.receiveRoutes = function * (payload, sender) {
   let gotNewRoute = false
   for (const route of routes) {
     // We received a route from another connector, but that connector
-    // doesn't actually belong to the route, so ignore it.
+    // doesn't actually appear as the first hop of the route, so ignore it.
```

Comment:
```if (route.source_account !== sender) continue```

`sender` is the address of the connector that forwarded us the route. If `route.source_account` doesn't match it, then we assume that the connector sent us a random route and we ignore it. 
In other words., we verify that the _route_ belongs to the _connector_ that broadcasted it (the code comment is backwards, though you could probably argue it either way).